lib/xenops_server.ml

```diff
@@ -1400,13 +1401,12 @@ and perform ?subtask ?result (op: operation) (t: Xenops_task.t) : unit =
 			debug "VM.receive_memory %s power_state = %s" id (state.Vm.power_state |> rpc_of_power_state |> Jsonrpc.to_string);*)
 
 			(* set up the destination domain *)
-			perform_atomics (
-				simplify [VM_create (id, Some memory_limit);] @
-				(* Perform as many operations as possible on the destination domain before pausing the original domain *)
-				(atomics_of_operation (VM_restore_vifs id))
-			) t;
-
 			(try
+				perform_atomics (
```

Comment:
@robhoes The vm_receive is invoked on the destination host from the vm_migrate bit in the source host and the problem, that this ticket is trying to solve, is to handle the scenario where the vm_receive fails -  so that the vm_migrate gets a handshake.Error and triggers the exception. Otherwise, the connection never closes and eventually times out, thus timing out the xapi client call.